### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by yoshuawuyts**
setup github actions ci. Thanks!

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
Tests are failing, but that's more or less expected because #49 didn't upgrade all crates.

cc/ @tirr-c do you think this is good to merge, or should we only test the crates we know will pass?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  RUSTFLAGS: -Dwarnings
+
+jobs:
+  build_and_test:
+    name: Build and test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        rust: [nightly]
```
Comment (tirr-c) [inline .github/workflows/ci.yaml]: We have async/await on stable, so why not test on it? 😜 
```suggestion
        rust: [stable, nightly]
```

I wonder if we have a way to make nightly tests not affect the CI status.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  check_fmt_and_docs:
+    name: Checking fmt and docs
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+
+    - id: component
+      uses: actions-rs/components-nightly@v1
+      with:
+        component: rustfmt
```
Comment (tirr-c) [inline .github/workflows/ci.yaml]: It seems `actions-rs/components-nightly` is [deprecated], but... we don't need to hang on nightly!

[deprecated]: https://github.com/actions-rs/components-nightly#deprecation-notice

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    - uses: actions-rs/toolchain@v1
+      with:
+          toolchain: ${{ steps.component.outputs.toolchain }}
```
Comment (tirr-c) [inline .github/workflows/ci.yaml]:
```suggestion
          toolchain: stable
```

**Review by tirr-c — COMMENTED** [review]
CI errors are a bit noisy, but I think it's good to merge! We can work toward the green CI status.

Some nits:

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (yoshuawuyts) [reply .github/workflows/ci.yaml]: okay, moved to stable. Trying again!

**Pull request merged**